juju/juju
# [JUJU-860] SystemState to return error instead of nil if pool is closed

#### Description

Currently the `SystemState` function in `state/pool.go` returns a `nil` whenever the pool is closed. This PR provides a semantics preserving (as mechanical as possible) change to have the `SystemState` function to return an error instead of a nil when the pool is closed.

Related [LP bug](https://bugs.launchpad.net/juju/+bug/1967330) reports that a client is getting a nil pointer de-refence seemingly coming from the `SystemState`. This change doesn't necessarily fix the LP bug by itself, but it's a step towards it. Regardless of that bug, we should have that function to return an error instead of nil if the pool is closed.

#### QA Steps

The CI tests should pass and the following should work fine

```sh
  $ juju bootstrap localhost test
  $ juju upgrade-controller --build-agent
```

as well as with `juju enable-ha`, e.g. :

```sh
  $ juju bootstrap localhost test
  $ juju enable-ha
  $ juju upgrade-controller --build-agent
```

should all be working ok.
